homeassistant/components/sighthound/image_processing.py

```diff
@@ -109,17 +123,27 @@ def fire_person_detected_event(self, person):
 
     def save_image(self, image, people, directory):
         """Save a timestamped image with bounding boxes around targets."""
-        img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+        try:
+            img = Image.open(io.BytesIO(bytearray(image))).convert("RGB")
+        except UnidentifiedImageError:
+            _LOGGER.warning("Unable to process image, bad data")
+            return
         draw = ImageDraw.Draw(img)
 
         for person in people:
             box = hound.bbox_to_tf_style(
                 person["boundingBox"], self._image_width, self._image_height
             )
             draw_box(draw, box, self._image_width, self._image_height)
+
         latest_save_path = directory / f"{self._name}_latest.jpg"
         img.save(latest_save_path)
 
+        if self._save_timestamped_file:
+            timestamp_save_path = directory / f"{self._name}_{self._last_detection}.jpg"
+            img.save(timestamp_save_path)
+            _LOGGER.info("Sighthound saved file %s", timestamp_save_path)
```

Comment:
Is this message for the user or for the developer?

Reply:
The user

Reply:
Ok.